plugins/experimental/access_control/plugin.cc

```diff
@@ -415,12 +415,18 @@ contHandleAccessControl(const TSCont contp, TSEvent event, void *edata)
 
           TSHandleMLocRelease(serverRespBufp, TS_NULL_MLOC, serverRespHdrLoc);
         } else {
-          AccessControlError("failed to retrieve server response header");
+          int len;
+          char *url = TSHttpTxnEffectiveUrlStringGet(txnp, &len);
+          AccessControlError("failed to retrieve server response header for request url:%.*s", (len ? len : 7),
```

Comment:
Not liking the naked "7" there. I would probably put a `static const std::string_view UNKNOWN{"unknown"};` at the top of the function and then use `UNKNOWN.size()` and `UNKNOWN.data()` in the error messages.

Alternatively, there is BufferWriter support for this. You could do
```
ts::LocalBufferWriter<4096> w;
w.print("Failed to retrieve server response header for request url: '{}'", ts::bwf::FirstOf(std::string_view{url, len}, "unknown");
AccessControlError("%.*s", int(w.size()), w.data());
```
It's a bit awkward because there are different APIs involved, but if `AccessControlError` had an overload that was BufferWriter based it would be cleaner.